Source/WebKit/WPEPlatform/wpe/drm/WPEDRM.cpp

```diff
@@ -86,6 +86,11 @@ bool Crtc::modeIsCurrent(drmModeModeInfo* mode) const
     return !memcmp(&m_currentMode.value(), mode, sizeof(drmModeModeInfo));
 }
 
+void Crtc::setCurrentMode(drmModeModeInfo* mode)
```

Comment:
I have a controversial suggestion here. I see that `modeIsCurrent` and `setCurrentMode` take pointers to `mode`, but then treat them as never null. No null check, no assertions: they trust that the caller has done them.

However, `memcmp` would, I think, result in undefined behaviour for a non-zero size but null `mode`. Similarly here, if we de-reference a null `mode`.

I think these two cases should be const references, even at the cost of doing `&mode` in `memcmp`, just so that we're making the semantics explicit: they expect non-null pointers.

Reply:
All the more reason to change the class API, that's what I mean.